torchtune/models/flux/_model_builders.py

```diff
@@ -67,7 +69,154 @@ def flux_1_autoencoder(
+        depth_single_blocks=38,
+        axes_dim=[16, 56, 56],
+        theta=10_000,
+        qkv_bias=True,
+        use_guidance=True,
+    )
+
+
+def flux_1_schnell_flow_model():
+    """
+    Flow-matching model for FLUX.1-schnell
+
+    Returns:
+        FluxFlowModel
+    """
+    return FluxFlowModel(
+        in_channels=64,
+        out_channels=64,
+        vec_in_dim=768,
+        context_in_dim=4096,
+        hidden_size=3072,
+        mlp_ratio=4.0,
+        num_heads=24,
+        depth=19,
+        depth_single_blocks=38,
+        axes_dim=[16, 56, 56],
+        theta=10_000,
+        qkv_bias=True,
+        use_guidance=False,
+    )
+
+
+def lora_flux_1_dev_flow_model(
+    lora_rank: int,
+    lora_alpha: float,
+    lora_dropout: float = 0.0,
+    quantize_base: bool = False,
+    use_dora: bool = False,
+):
+    """
+    Flow-matching model for FLUX.1-dev with linear layers replaced with LoRA modules.
+
+    Args:
+        lora_rank (int): rank of each low-rank approximation
+        lora_alpha (float): scaling factor for the low-rank approximation
+        lora_dropout (float): dropout probability for the low-rank approximation. Default: 0.0
+        quantize_base (bool): Whether to quantize base model weights. Default: False
+        use_dora (bool): Decompose the LoRA weight into magnitude and direction, as
+            introduced in "DoRA: Weight-Decomposed Low-Rank Adaptation" (https://arxiv.org/abs/2402.09353).
+            Default: False
+
+    Returns:
+        FluxFlowModel
+    """
+    model = flux_1_dev_flow_model()
+    _replace_linear_with_lora(
+        model,
+        rank=lora_rank,
+        alpha=lora_alpha,
+        dropout=lora_dropout,
+        quantize_base=quantize_base,
+        use_dora=use_dora,
+    )
+    return model
+
+
+def lora_flux_1_schnell_flow_model(
+    lora_rank: int = 16,
+    lora_alpha: float = 16.0,
+    lora_dropout: float = 0.0,
+    quantize_base: bool = False,
+    use_dora: bool = False,
+):
+    """
+    Flow-matching model for FLUX.1-schnell with linear layers replaced with LoRA modules.
+
+    Args:
+        lora_rank (int): rank of each low-rank approximation
+        lora_alpha (float): scaling factor for the low-rank approximation
+        lora_dropout (float): dropout probability for the low-rank approximation. Default: 0.0
+        quantize_base (bool): Whether to quantize base model weights. Default: False
+        use_dora (bool): Decompose the LoRA weight into magnitude and direction, as
+            introduced in "DoRA: Weight-Decomposed Low-Rank Adaptation" (https://arxiv.org/abs/2402.09353).
+            Default: False
+
+    Returns:
+        FluxFlowModel
+    """
+    model = flux_1_schnell_flow_model()
+    _replace_linear_with_lora(
+        model,
+        rank=lora_rank,
+        alpha=lora_alpha,
+        dropout=lora_dropout,
+        quantize_base=quantize_base,
+        use_dora=use_dora,
+    )
+    return model
+
+
+def _replace_linear_with_lora(
```

Comment:
👀 @felipemello1

Reply:
I think we've talked about this one for a bit anyways, so great to see it finally materializing. Maybe let's put in `modules/peft/_utils.py` or something?

Reply:
One other thing we'll have to watch out for here.. for NF4-quantized layers we also add state dict hooks to prevent higher memory on checkpoint save. It may be worth including that logic in this utility